### ISSUE TITLE
Steam flatpak is not able to launch Retroarch, which is available in steam as an Appimage  

### ISSUE THREAD

**Issue by tjssoldier** [flathub/com.valvesoftware.Steam#770]
**Game information**
Retroarch - Steam Version

**Distribution name and version where applicable**
Linux Mint 19.3 Cinnamon

**Flatpak info**
 Flatpak version: Flatpak 1.11.1
 flatpak --gl-drivers: default host

**Problem description**
The Steam Flatpak version is no able to launch Retroarch, which is available in steam as an Appimage  

**Does this issue reproduce with native Steam**
No, in native steam works as expected

**Comment by nanonyme** [comment]
I'm assuming AppImage simply cannot work inside flatpak.

**Comment by gasinvein** [comment]
Yes, AppImages use fuse to mount themselves, so it's not possible to run them in flatpak (fuse requires suid root binary `fusermount`).
I wonder what was the reason for them to ship AppImage through Steam, though.

**Comment by tjssoldier** [comment]
So there is no hope to get it working ? I really like the idea of using Steam as a flatpak.

**Comment by gasinvein** [comment]
Well, one terrible workaround would be allowing the Steam app executing commands on host, and adding a wrapper for `fusermount` that would run one on host. But this would effectively eliminate the sandbox from security perspective, and it's probably not what you want (and not something we could ship by default). So no, I'd consider it impossible to run appimages directly in flatpak sandbox.

RetroArch itself is [available](https://flathub.org/apps/details/org.libretro.RetroArch) from Flathub, though.

**Comment by Taulim** [comment]
> So there is no hope to get it working ? I really like the idea of using Steam as a flatpak.

I did manage to make it work forcing it to run under Proton (in my case proton-ge) :smile: 
It download and install the windows version instead of the AppImage file.

**Comment by tjssoldier** [comment]
Maybe a solution ?

https://www.libretro.com/index.php/retroarch-steam-important-information-for-linux-users/

It seems they are changing from the Appimage to another format.

**Comment by tjssoldier** [comment]
> > So there is no hope to get it working ? I really like the idea of using Steam as a flatpak.
> 
> I did manage to make it work forcing it to run under Proton (in my case proton-ge) smile It download and install the windows version instead of the AppImage file.

Yes, but it is better use the Linux native version.

**Comment by tjssoldier** [comment]
It seems that there is a way to implement support for Appimages inside flatpaks. I found this pull request in lutris flatpak repo:

https://github.com/flathub/net.lutris.Lutris/pull/201#issue-1090262395

But i not a dev, so i don't know if can be implemented in Steam flatpak

**Comment by gasinvein** [comment]
@tjssoldier There is a way indeed, I've mentioned it [here](https://github.com/flathub/com.valvesoftware.Steam/issues/770#issuecomment-859962598) as a "terrible workaround". And this is exactly what flathub/net.lutris.Lutris#201 does.

**Comment by gasinvein** [comment]
RetroArch devs have got rid of AppImage; looks like they're now using Steam Linux Runtime container (which bumps flatpak version requirement to 1.12+, similarly to #642). So I believe the original issue is resolved. However, they're not the only devs to release appimage-packed game on Steam for whatever reason, so I'm going to convert this issue into a more generic one.

**Comment by gasinvein** [comment]
See also flatpak/xdg-desktop-portal#695

**Comment by Bohmthedudevulture** [comment]
@gasinvein Entschuldigung, but what about implement a fake fusermount that calls the real one in the in the background, but with a firewall?

**Comment by gasinvein** [comment]
@Bohmthedudevulture I'm not sure what a "firewall" means in this context. Anyway, calling the real fusermount means escaping the sandbox.

**Comment by soredake** [comment]
Actually, appimages can be run in flatpak, they can be launched with https://github.com/AppImage/AppImageKit/issues/841 argument and [NO_CLEANUP=1](https://github.com/AppImage/AppImageKit/pull/842/files#diff-50977a57b5cd77600b845d12867637b22f364a61b16a6d80791037caf018429fR665-R669) env to skip cleaning up files after each run, but it's need to be done by user manually for all appimage games.

**Comment by HugKitten** [comment]
For those who want to get this to work, there is a solution. I'll use OVR as an example. First make sure you are not using the proton version, then find the run.sh file (maybe be named something else) and edit it to something like this. Make sure to replace OVRAS.AppImage with the path and name of the app image:

```#!/usr/bin/env bash
unset LD_LIBRARY_PATH
./OVRAS.AppImage --appimage-extract-and-run

export APPIMAGE_EXTRACT_AND_RUN=1
./my.AppImage [...]

export APPIMAGE_EXTRACT_AND_RUN=1
env NO_CLEANUP=1 ./my.AppImage```
